libs/utils/analysis/latency_analysis.py

```diff
@@ -0,0 +1,361 @@
+# Configure logging
+import logging
+
+# Tuple representing all IDs data of a Task
+TaskData = namedtuple('TaskData', ['pid', 'names', 'label'])
+
+class LatencyAnalysis(AnalysisModule):
+    """
+    Support for plotting Latency Analysis data
+
+    :param trace: input Trace object
+    :type trace: :mod:`libs.utils.Trace`
+    """
+
+    def __init__(self, trace):
+        super(LatencyAnalysis, self).__init__(trace)
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+
+    @memoized
+    def _dfg_latency_df(self, task):
+
+        if not self._trace.hasEvents('sched_wakeup'):
+            logging.warn('Events [sched_wakeup] not found, '
+                         'cannot compute CPU active signal!')
+            return None
+        if not self._trace.hasEvents('sched_switch'):
+            logging.warn('Events [sched_switch] not found, '
+                         'cannot compute CPU active signal!')
+            return None
+
+        # Get task data
+        td = self._getTaskData(task)
+        if not td:
+            return None
+
+        wk_df = self._dfg_trace_event('sched_wakeup')
+        sw_df = self._dfg_trace_event('sched_switch')
+
+        # Filter Task's WAKEUP events
+        task_wakeup = wk_df[wk_df.pid == td.pid][['success', 'target_cpu', 'pid']]
+
+        # Filter Task's START events
+        task_events = (sw_df.prev_pid == td.pid) | (sw_df.next_pid == td.pid)
+        task_switches_df = sw_df[task_events]\
+            [['__cpu', 'prev_pid', 'next_pid', 'prev_state']]
+
+        # Unset prev_state for switch_in enents, i.e.
+        # we don't care about the status of a task we are replacing
+        task_switches_df.prev_state = task_switches_df.apply(
+            lambda r : np.nan if r['prev_pid'] != td.pid else r['prev_state'], axis = 1)
+
+        # Rename prev_state
+        task_switches_df.rename(columns={'prev_state' : 'curr_state'}, inplace=True)
+
+        # Fill in Running status
+        task_switches_df.curr_state = task_switches_df.curr_state.fillna(value='A')
+
+        # Join Wakeup and SchedSwitch events
+        task_latency_df = task_wakeup.join(task_switches_df, how='outer',
+                                             lsuffix='_wkp', rsuffix='_slp')
+        # Remove not required columns
+        task_latency_df = task_latency_df[['target_cpu', '__cpu', 'curr_state']]
+        # Set Wakeup state on each Wakeup event
+        task_latency_df.curr_state = task_latency_df.curr_state.fillna(value='W')
+
+        # Forward annotate task state
+        task_latency_df['next_state'] = task_latency_df.curr_state.shift(-1)
+
+        # Forward account for previous state duration
+        task_latency_df['t_start'] =  task_latency_df.index
+        task_latency_df['t_delta'] = (
+            task_latency_df['t_start'].shift(-1)
+            - task_latency_df['t_start']
```

Comment:
yeah... that was intentional to better show what we are doing... actually I'll make it even more strange formatted by aligning the two task_latency_df